### DISCUSSION TITLE
3DS QR code using wrong IP address

### DISCUSSION THREAD

**Discussion by the0remora** [retronas/discussions#72] [Q&A]
I ran into an issue with the /opt/retronas/scripts/3ds_qr.sh file. I assign addresses via DHCP and usually have PfSense set up with a Static Assignment. At some point during testing I think I used the WiFi on my Pi4 instead of the LAN and it got the address 172.17.0.0 it wrote that into the 3ds_qr.sh file and never updated it (LAN uses 172.19.0.0). Could some way be implemented RetroNAS to force an update either periodically or during reboot or as part of one of the update scripts?

TLDR: RetroNAS got an IP address during testing, that address has changed but the old one is still used by scripts.

Edit replaced * in IP address with 0

**Comment by sairuk** [comment]
Not really an issue since the script worked as defined, am converting to discussion to continue with enhancement request conversation

**Reply by Defkyle** [reply]
I actually believe this is an issue. Even after deleting all the QR codes with the previous IP address and running the script again the IP on the QR code stays the same. Shouldn't the script look at the current IP address?

**Reply by sairuk** [reply]
The enhancement request was in reference to the periodic update.

If the device had the wrong ip when installing the script then a reinstall when on the correct ip will update the script inplace, so I don't see anything wrong in that regard, wrong ip is wrong ip but if you are still having issues lets see if we can work out whats going on.

Did you reinstall the 3DS solution after you put the device on the correct ip address? 

Just deleting the qr codes is not enough to correct the ip address since the address is written out as part of the installation, we do not necessarily have to take this approach but right now that is the functionality.

You can confirm the script ip address by looking at the `MYURL` entry in `/opt/retronas/scripts/3ds_gr.sh` e.g. `grep MYURL /opt/retronas/scripts/3ds_gr.sh` in the terminal, if it is incorrect please reinstall the 3ds solution

**Reply by Defkyle** [reply]
Thanks you, I just changed the IP address manually in the script and now it's fine. 
Thank you!

**Reply by the0remora** [reply]
Relying on the user to know they need to perform a reinstall is not a user friendly solution. I knew enough to hunt down `/opt/retronas/scripts/3ds_gr.sh` and Defkyle also understood how to access and edit that file. There may already have been someone else who has only seen the RetroRGB announcement video who managed to get RetroNAS setup and abandoned it because this didn't work as expected.

I do not believe that the qrcodes should be updated on the fly but DHCP is dynamic. Just because RetroNAS had the address *. *. *.124 when MYURL was generated does not mean it stayed that. In my case I switched from WiFi (testing RetroNAS) to wired connection. This changed the ip address needed.

Is it possible to create the variable `CURRENTIP` and have that part of `MYURL`? `CURRENTIP` could be created in boot or when qrcodes are generated.

**Reply by sairuk** [reply]
We are looking at environment vars as a potential solution for semi-dynamic content, so yes this is on the table for discussion.
